spec/requests/products/show_spec.rb

```diff
@@ -0,0 +1,61 @@
+require "web_spec"
+
+RSpec.describe "GET /products/:id", type: :request do
+  context "when product with given id does not exist" do
+    before do
+      do_request(0)
```

Comment:
Well, I treated the `0` as an argument for `do_request` to test the request for non-existent product. Do you mean it's a bad idea in general, or just that it's not very clear in the code right now?
If it's the latter, will something like `non_existent_product_id = 0` and then `do_request(non_existent_product_id)` be better?